tests/asgi/tests.py

```diff
@@ -319,3 +353,61 @@ async def test_concurrent_async_uses_multiple_thread_pools(self):
         self.assertEqual(len(sync_waiter.active_threads), 2)
 
         sync_waiter.active_threads.clear()
+
+    async def test_asyncio_cancel_error(self):
+        application = ASGIHandlerWithEditedScope()
+        # Construct HTTP request.
+        scope = self.async_request_factory._base_scope(
+            path="/", asyncio_cancel_error=False
+        )
+        communicator = ApplicationCommunicator(application, scope)
+        await communicator.send_input({"type": "http.request"})
+        await communicator.receive_output()
+        self.assertEqual(scope["asyncio_cancel_error"], True)
+        # Give response.close() time to finish.
+        await communicator.wait()
+
+
+class ASGIHandlerWithEditedScope(ASGIHandler):
```

Comment:
Hello @carltongibson , I apologize for the lack of updates from me over the past few days. We had an issue with power in my town.

You mentioned using a variable in the `scope` to flag True if the pending task has been canceled and `asyncio.cancelError` is encountered. Where should this happen? I subclassed `ASGIHandler` here and polymorphed `handle` method to allow for the check. This is just a rough idea, am hoping to get more reviews on how to properly test for the canceled pending task.